src/main/java/org/apache/maven/shared/utils/logging/MessageUtils.java

```diff
@@ -201,4 +201,21 @@ public void run()
             Runtime.getRuntime().addShutdownHook( shutdownHook );
         }
     }
+
+    /**
+     * Get the terminal width or 0 if the width can not be determined.
+     *
+     * @return the terminal width
+     */
+    public static int getTerminalWidth()
```

Comment:
This needs a test.

Reply:
There's no way to easily ensure the test is run from inside a terminal. This would always fail when running from an IDE for example.

Reply:
I've added a unit test, but I'm a bit skeptical as it's really tied to Jansi's internal and is mostly about checking that `getTerminalWidth()` returns the value that will be returned by the underlying implementation.